Automattic/pocket-casts-android
# Fix the podcast settings fragment leak

## Description

While testing the app with Leak Canary when you open the podcast settings and return to the podcast page it reports a leak. This change fixes this leak. The cause of the issues seems to be setting the toolbar in the activity, I don't think this is needed so I have removed it.

This PR contains some other clean up

## Testing Instructions
1. In base.gradle uncomment `debugImplementation androidLibs.leakCanary`
2. Setup a filter in logcat `package:mine LeakCanary`
3. Go to the Podcasts tab
4. Open a podcast page
5. Tap the settings cog 
6. Wait for the log message and ignore `Found 1 object retained` message from the Podcast Fragment
7. Tap the back button in the toolbar
9. ✅ Verify if there is an object retained log line that there is also a `All retained objects have been garbage collected` shortly after.
